src/sdl2_wrapper.cpp

```diff
@@ -459,6 +459,40 @@ CAMLprim value resdl_SDL_GetDesktopDisplayMode(value vDisplay) {
   CAMLreturn(ret);
 };
 
+CAMLprim value resdl_SDL_GetDisplayBounds(value vDisplay) {
+  CAMLparam1(vDisplay);
+  CAMLlocal1(rect);
+
+  int displayIndex = Int_val(vDisplay);
+  SDL_Rect sdlRect;
+
+  SDL_GetDisplayBounds(displayIndex, &sdlRect);
+
+  rect = caml_alloc(4, 0);
+  Store_field(rect, 0, sdlRect.x);
```

Comment:
These should be `Store_field(rect, 0, Val_int(sdlRect.x))`, I believe

Reply:
Changing all of those seems to fix the crash 👍

Reply:
It isn't with the display mode functions above. But still certainly worth a try.

Reply:
Yep, that was it. Thanks! :tada: